scipy/interpolate/interpolate.py

```diff
@@ -1393,7 +1393,9 @@ def from_derivatives(cls, xi, yi, orders=None, extrapolate=None):
         if orders is None:
             orders = [None] * m
         else:
-            if isinstance(orders, integer_types):
+            if hasattr(orders, '__iter__'):
```

Comment:
as a matter of style, I'd still prefer to check for an integer: either an instance of six.integer_types or something with a dtype being a subdtype of `np.integer`, gh-4106 or equivalent